sihl/src/database_migration_repo.ml

```diff
@@ -23,79 +23,91 @@ end
 module type Sig = sig
   module Migration = Migration
 
-  val create_table_if_not_exists : unit -> unit Lwt.t
-  val get : namespace:string -> Migration.t option Lwt.t
-  val get_all : unit -> Migration.t list Lwt.t
-  val upsert : state:Migration.t -> unit Lwt.t
+  val create_table_if_not_exists : string -> unit Lwt.t
+  val get : string -> namespace:string -> Migration.t option Lwt.t
+  val get_all : string -> Migration.t list Lwt.t
+  val upsert : string -> Migration.t -> unit Lwt.t
 end
 
+(* Common functions *)
+let get_request table =
+  Caqti_request.find_opt
+    Caqti_type.string
+    Caqti_type.(tup3 string int bool)
+    (Format.sprintf
+       {sql|
+       SELECT
+         namespace,
+         version,
+         dirty
+       FROM %s
+       WHERE namespace = ?;
+       |sql}
+       table)
+;;
+
+let get table ~namespace =
+  Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
+      Connection.find_opt (get_request table) namespace
+      |> Lwt.map Database.raise_error)
+  |> Lwt.map (Option.map Migration.of_tuple)
+;;
+
+let get_all_request table =
+  Caqti_request.collect
+    Caqti_type.unit
+    Caqti_type.(tup3 string int bool)
+    (Format.sprintf
+       {sql|
+       SELECT
+         namespace,
+         version,
+         dirty
+       FROM %s;
+       |sql}
+       table)
+;;
+
+let get_all table =
+  Database.query (fun (module Connection : Caqti_lwt.CONNECTION) ->
+      Connection.collect_list (get_all_request table) ()
+      |> Lwt.map Database.raise_error)
+  |> Lwt.map (List.map Migration.of_tuple)
+;;
+
 module MariaDb : Sig = struct
   module Migration = Migration
 
-  let create_request =
+  let create_request table =
     Caqti_request.exec
       Caqti_type.unit
-      {sql|
-       CREATE TABLE IF NOT EXISTS core_migration_state (
+      (Format.sprintf
+         {sql|
+       CREATE TABLE IF NOT EXISTS %s (
```

Comment:
Could this be an attack vector for SQL injection?

Reply:
No, the table name is not based on any end user input. You need access to the env variables to set it.